Functions/Mock.ps1

```diff
@@ -1544,3 +1547,75 @@ function Remove-MockFunctionsAndAliases {
         & $script:SafeCommands['Remove-Item'] "function:/$($bootstrapFunction.Name)"
     }
 }
+
+
+function Repair-ConflictingParameters {
+    [CmdletBinding()]
+    [OutputType([System.Management.Automation.CommandMetadata])]
+    param(
+        [Parameter(Mandatory)]
+        [System.Management.Automation.CommandMetadata]
+        $Metadata
+    )
+
+    $paramMetadatas = @()
+    $paramMetadatas += $Metadata.Parameters.Values
+
+    foreach ($paramMetadata in $paramMetadatas) {
+        if ($paramMetadata.IsDynamic) {
+            continue
+        }
+
+        $conflictingParams = Get-ConflictingParameterNames
+        if ($conflictingParams -contains $paramMetadata.Name) {
+            $paramName = $paramMetadata.Name
+            $newName = "$(Get-ParameterPrefixForConflictingParameters)$paramName"
+            $paramMetadata.Name = $newName
+            $paramMetadata.Aliases.Add($paramName)
+
+            $Metadata.Parameters.Remove($paramName) | Out-Null
+            $Metadata.Parameters.Add($newName, $paramMetadata)
+        }
+    }
+
+    $Metadata
+}
+
+function Reset-ConflictingParameters {
+    [CmdletBinding()]
+    [OutputType([hashtable])]
+    param(
+        [Parameter(Mandatory)]
+        [hashtable]
+        $BoundParameters
+    )
+
+    $parameters = $BoundParameters.Clone()
+    $prefix = Get-ParameterPrefixForConflictingParameters
+
+    Get-ConflictingParameterNames |
```

Comment:
To avoid conflict with mocked `Where-Object` and `Foreach-Object` you should use the commands from the SafeCommands table, but using pipeline is slow anyway. So a better alternative is to use foreach (keyword) and if, which solves both problems at the same time. Please rewrite it.

Reply:
Done